ct-app/postman/postman_tasks.py

```diff
@@ -81,12 +90,34 @@ async def async_send_1_hop_message(
     )
 
     # node is reachable, messages can be sent
-    for _ in range(count):
-        state = await api.send_message(address, "foo", [peer_id])  # noqa: F841
+    for idx in range(count):
+        state = await api.send_message(address, "foo", [peer_id])
+
+        if not state:
+            log.error("Could not send message")
+            break
+    else:
+        log.info(
+            f"{count} messages sent to `{peer_id}` "
+            + f"via {node_list[node_index]}(#{node_index} - {api_host})"
+        )
 
-    log.info(
-        f"{count} messages sent to `{peer_id}` "
-        + f"via {node_list[node_index]}(#{node_index} - {api_host})"
+        return "SUCCESS"
+
+    log.error("Could not send all messages")
+
+    node_address, node_index = get_next_node_address(node_list, node_index)
+
+    if not node_address:
+        log.info("This is the last node in the list, stopping")
+        log.info(f"Only sent {idx} messages")
+        return "FAIL"
+
+    log.info(f"Redirecting task to {node_address} (#{node_index} - {api_host})")
+    app.send_task(
+        f"{envvar('TASK_NAME')}.{node_address}",
+        args=(peer_id, count - idx, node_list, node_index),
+        queue=node_address,
     )
 
-    return "SUCCESS"
+    return "SPLITTED"
```

Comment:
These return values seem to be part of a protocol. If so, they should be defined separately, in a class (DataClass?)

Reply:
Agreed. Resolved in new commit.